### PR TITLE
fix: Use and enforce structured logging. Fixes #7243 

### LINKED ISSUE #7243: Improve taskset/HTTP template logging
A lot of the logging for taskset/http template logs the Go object, which results in log lines that are (a) not very human readable and (b) not very machine readable, therefore less helpful that they could be:

Example:

```
             controller | time="2021-11-17T18:01:51.237Z" level=debug msg="patching the exiting taskset" TaskSet="map[controller-plugin-template-qj4gt:{main {[] []} {[] [] <nil> <nil>} map[] nil {map[] map[]} <nil> [] nil nil nil nil nil nil nil nil 0xc0005b5308 [] [] [] nil <nil> nil <nil> <nil> []   <nil>  <nil> nil [] nil  nil nil nil }]" namespace=argo workflow=controller-plugin-template-qj4gt
```

We could log the JSON of the object, but that's not machine readable. Better to just be structured.

### PR DESCRIPTION
**Pull request by alexec**
Fixes #7243

### CODE REVIEW

File: workflow/controller/agent.go [hunk 1]
```diff
@@ -112,11 +115,12 @@ func (woc *wfOperationCtx) createAgentPod(ctx context.Context) (*apiv1.Pod, erro
...
-					Name:    "main",
-					Command: []string{"argoexec"},
-					Args:    []string{"agent"},
-					Image:   woc.controller.executorImage(),
-					Env:     envVars,
+					Name:            "main",
+					Command:         []string{"argoexec"},
+					Args:            []string{"agent"},
+					Image:           woc.controller.executorImage(),
+					ImagePullPolicy: woc.controller.executorImagePullPolicy(),
```
Comment (alexec) [inline workflow/controller/agent.go:122]: this also needed fixing

File: hack/check-logging.sh [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+#
+# I've tried hard to think of any occasion when I'd prefer unstructured logging. I can't think of any times, but here
+# might be some edge cases.
+#
+# As a last resort, use `log.Info(fmt.Sprintf(""))`.
+
+set -eu
+
+from=$(git merge-base --fork-point origin/master)
+count=$(git diff "$from" -- '*.go' | grep '^+' | grep -v '\(fmt\|errors\).Errorf' | grep -c '\(Debug\|Info\|Warn\|Warning\|Error\)f' || true)
```
Comment (simster7) [inline hack/check-logging.sh:23]: Can also have `Info` (as opposed to `Infof`)

File: workflow/controller/agent.go (hunk 1 above)
Comment (simster7) [reply workflow/controller/agent.go:122]: Thanks!

File: hack/check-logging.sh (hunk 2 above)
Comment (alexec) [reply hack/check-logging.sh:23]: um, it disallows `Infof` but allows` Info`

**Comment by alexec** [comment]
@sarabala1979 @simster7 bump!

**Review by sarabala1979 — APPROVED** [review]

**Pull request merged**